### PR TITLE
feat: pox-4 support

### PR DESCRIPTION
**Pull request by zone117x**
* Force-unlock / sunsetting code for pox-3 locks
* Ingestion for pox-4 events
* Initial ingestion for `revoke-delegate-stx`

Tests for pox-4 have been implemented in a separate PR to make things easier to review: https://github.com/hirosystems/stacks-blockchain-api/pull/1762

### CODE REVIEW

**Review by rafaelcr — APPROVED** [review]
Nice!

File: src/api/controllers/db-controller.ts [hunk 1]
```diff
@@ -315,8 +314,17 @@ export function parsePox2Event(poxEvent: DbPox2Event | DbPox3Event) {
         },
       };
     }
+    case SyntheticPoxEventName.RevokeDelegateStx: {
+      return {
+        ...baseInfo,
+        data: {
+          amount_ustx: poxEvent.data.amount_ustx.toString(),
+          delegate_to: poxEvent.data.delegate_to,
```
Comment (janniks) [inline src/api/controllers/db-controller.ts:322]: I'm curious where this part is in the stacks-core codebase (haven't been able to find, would love to peek at that code)

Comment (zone117x) [reply src/api/controllers/db-controller.ts:322]: The synthetic pox events are at https://github.com/stacks-network/stacks-core/blob/next/pox-locking/src/events.rs
This `revoke` one is WIP https://github.com/stacks-network/stacks-core/pull/4157

File: src/datastore/common.ts [hunk 2]
```diff
@@ -384,67 +386,75 @@ export interface DbPox2DelegateStackStxEvent extends DbPox2BaseEventData {
...
-  | DbPox2DelegateStackExtendEvent
-  | DbPox2StackAggregationCommitEvent
-  | DbPox2StackAggregationCommitIndexedEvent
-  | DbPox2StackAggregationIncreaseEvent;
+export interface DbPoxSyntheticRevokeDelegateStxEvent extends DbPoxSyntheticBaseEventData {
+  name: SyntheticPoxEventName.RevokeDelegateStx;
+  data: {
+    // TODO: determine what data is available for this event type
+    amount_ustx: bigint;
+    delegate_to: string;
```
Comment (janniks) [inline src/datastore/common.ts:445]: I'm assuming only the sender address is available (given pox clar code), but not sure how much more the node will provide.

Comment (zone117x) [reply src/datastore/common.ts:445]: The WIP PR https://github.com/stacks-network/stacks-core/pull/4157 is likely to add more data. There will probably be a follow-up API PR once that is ready

**Review by janniks — APPROVED** [review]
Looking good 👍

**Pull request merged**